File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.py
'''
수를 기준으로 정렬 -> 가장 앞의 수를 기준으로 우선적으로 정렬되어야함

'''

def solution(numbers):
    
    numbers_str = list(map(str, numbers))
                      
    # 가장 앞의 수를 기준으로 정렬 -> 문자열 곱하기! -> 같은 문제열을 n번 반복
    numbers_str.sort(key = lambda x : (x*3), reverse=True)
    return str(int(''.join(numbers_str)))

File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.java
import java.util.*;

class Solution {
    
    
    public String solution(int[] numbers) {
        
        String[] numbersStr = new String[numbers.length];
        for (int i = 0; i < numbers.length; i++) {
            numbersStr[i] = String.valueOf(numbers[i]);
        }

        Arrays.sort(numbersStr, (a,b) -> (b+b+b).compareTo(a+a+a));
        
        if(numbersStr[0].equals("0")){
            return "0";
        }

        StringBuilder result = new StringBuilder();
        for(int i=0; i<numbersStr.length; i++){
            result.append(numbersStr[i]);
        }

        return result.toString();
    }
}